File: javascript/toy-problems/scrolling_text.js
// Let's create some scrolling text!

// Your task is to complete the function which takes a string, and returns an array with all possible rotations of the given string, in uppercase.

// Example
// scrollingText("codewars") should return:

// [ "CODEWARS",
//   "ODEWARSC",
//   "DEWARSCO",
//   "EWARSCOD",
//   "WARSCODE",
//   "ARSCODEW"
//   "RSCODEWA",
//   "SCODEWAR" ]


let scrollingText = text => {
let answerArr = []
let newWord = text.toUpperCase().split('')
answerArr.push(text.toUpperCase())
for(i = 0; i < text.split('').length; i++) {
  if(newWord !== text) {
    let tempWord = newWord.shift() //C
    let temp = newWord.push(tempWord)
    answerArr.push(newWord.join(''))
  }
}
answerArr.pop()
return answerArr
}

console.log(scrollingText("codewars"));


//function scrollingText(text){
//   let result = []
  
//   for( var i = 0; i < text.length; i++){
//     result.push((text.slice(i) + text.slice(0,i)).toUpperCase())
//   }
  
//   return result
// }

File: python/toy-problems/scrolling_text.py
# Let's create some scrolling text!

# Your task is to complete the function which takes a string, and returns an array with all 
# possible rotations of the given string, in uppercase.

# Example
# scrollingText("codewars") should return:

# [ "CODEWARS",
#   "ODEWARSC",
#   "DEWARSCO",
#   "EWARSCOD",
#   "WARSCODE",
#   "ARSCODEW"
#   "RSCODEWA",
#   "SCODEWAR" ]

# My embarassing ass answer
def scrolling_text(text):
    new_list = []
    new_list.append(text.upper())

    temp_list = text[:1]
    second_list = text[1:]

    new_word = second_list + temp_list
    new_list.append(new_word.upper())
    for x in text:
        if new_word != text:
            temp_list = new_word[:1]
            second_list = new_word[1:]

            new_word = second_list + temp_list
            new_list.append(new_word.upper())

    return new_list[:-1]


scrolling_text("abcd")

# Better answer (Lol I'm retarded)
# def scrolling_text(text):
#     text = text.upper()
#     return [ text[i:] + text[:i] for i in range(len(text)) ]